framework/src/timeintegrators/ActuallyExplicitEuler.C

```diff
@@ -0,0 +1,236 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+// MOOSE includes
+#include "ActuallyExplicitEuler.h"
+#include "NonlinearSystem.h"
+#include "FEProblem.h"
+#include "PetscSupport.h"
+#include "LStableDirk4.h"
+
+// libMesh includes
+#include "libmesh/sparse_matrix.h"
+#include "libmesh/nonlinear_solver.h"
+#include "libmesh/preconditioner.h"
+
+registerMooseObject("MooseApp", ActuallyExplicitEuler);
+
+template <>
+InputParameters
+validParams<ActuallyExplicitEuler>()
+{
+  InputParameters params = validParams<TimeIntegrator>();
+
+  MooseEnum solve_type("consistent lumped lump_preconditioned", "consistent");
+
+  params.addParam<MooseEnum>(
+      "solve_type",
+      solve_type,
+      "The way to solve the system.  A 'consistent' solve uses the full mass matrix and actually "
+      "needs to use a linear solver to solve the problem.  'lumped' uses a lumped mass matrix with "
+      "a simple inversion - incredibly fast but may be less accurate.  'lump_preconditioned' uses "
+      "the lumped mass matrix as a preconditioner for the 'consistent' solve");
+
+  params.addClassDescription(
+      "Implementation of Explicit/Forward Euler without invoking any of the nonlinear solver");
+
+  return params;
+}
+
+/**
+ * Helper class to apply preconditioner
+ */
+class LumpedPreconditioner : public Preconditioner<Real>
+{
+public:
+  LumpedPreconditioner(const NumericVector<Real> & diag_inverse)
+    : Preconditioner(diag_inverse.comm()), _diag_inverse(diag_inverse)
+  {
+  }
+
+  virtual void init() override
+  {
+    // No more initialization needed here
+    _is_initialized = true;
+  }
+
+  virtual void apply(const NumericVector<Real> & x, NumericVector<Real> & y) override
+  {
+    y.pointwise_mult(_diag_inverse, x);
+  }
+
+protected:
+  /// The inverse of the diagonal of the lumped matrix
+  const NumericVector<Real> & _diag_inverse;
+};
+
+ActuallyExplicitEuler::ActuallyExplicitEuler(const InputParameters & parameters)
+  : TimeIntegrator(parameters),
+    MeshChangedInterface(parameters),
+    _solve_type(getParam<MooseEnum>("solve_type")),
+    _explicit_residual(_nl.addVector("explicit_residual", false, PARALLEL)),
+    _explicit_euler_update(_nl.addVector("explicit_euler_update", true, PARALLEL)),
+    _mass_matrix_diag(_nl.addVector("mass_matrix_diag", false, PARALLEL))
+{
+  _Ke_time_tag = _fe_problem.getMatrixTagID("TIME");
+
+  // Try to keep MOOSE from doing any nonlinear stuff
```

Comment:
Are you trying to say that people can actually override the solver behavior? Like through command-line supplied PETSc options?

Reply:
To me "Try to" sounds like there is a chance that the final solver type may not be the one specified here, i.e. LINEAR. If that's not the case, let's not mention the word "try".

Reply:
It's "try to"... because that setting is not foolproof.  MOOSE is built for doing nonlinear solves and it definitely could do something unnecessary for linear solves even with setting that option.  But at least we're going to "try" to tell it not to do that!

The comment is fine.